lib/matplotlib/cbook.py

```diff
@@ -361,9 +361,11 @@ class _BoundMethodProxy(object):
     Minor bugfixes by Michael Droettboom
     '''
     def __init__(self, cb):
+        self._hash = hash(cb)
+        self._callbacks = []
```

Comment:
Ah, long thread, so I naturally forgot to simply check the beginning...

I see why I never encountered this. It isn't that bound methods don't work
at all -- they do work just fine. The problem is that they are not getting
uniquely identified if they are connected multiple times, right?

My code that typically use callbacks would have a single "control" object
that would hold references to all the relevant objects and also perform
connections only once. Nothing ever went out of scope until the end of the
application, so I never noticed these issues. Disconnects also have always
worked for me because I always recorded the cids.


On Thu, Feb 19, 2015 at 10:15 AM, OceanWolf <notifications@github.com>
wrote:

> In lib/matplotlib/cbook.py
> <https://github.com/matplotlib/matplotlib/pull/4118#discussion_r24994287>:
>
> > @@ -361,9 +361,11 @@ class _BoundMethodProxy(object):
> >      Minor bugfixes by Michael Droettboom
> >      '''
> >      def __init__(self, cb):
> > +        self._hash = hash(cb)
> > +        self._callbacks = []
>
> @WeatherGod <https://github.com/WeatherGod> Sure, take a look at the
> second comment (#4118 (comment)
> <https://github.com/matplotlib/matplotlib/pull/4118#issuecomment-74774247>).
> This gives cases for testing. The (main) bug appears on the line where I
> write # Should *also* print 1 as it does not, it gives a different cid to
> the first, i.e. 2.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4118/files#r24994287>.
>